core/src/main/scala/org/bitcoins/core/gcs/SipHashKey.scala

```diff
@@ -0,0 +1,17 @@
+package org.bitcoins.core.gcs
+import scodec.bits.ByteVector
+case class SipHashKey(bytes: ByteVector) {
```

Comment:
One thing I would recommend doing here is extending `NetworkElement`

So the definition would look like this

```scala
case class SipHashKey(bytes: ByteVector) extends NetworkElement

Reply:
`NetworkElement` gives you a `.hex` and `.bytes` method you can call on to always get the byte/hex representation of the element. If you look at the definition of a bitcoin [`Transaction`](https://github.com/bitcoin-s/bitcoin-s/blob/master/core/src/main/scala/org/bitcoins/core/protocol/transaction/Transaction.scala#L19) you see it also extends `NetworkElement` which gives you a nice way to serialize the transaction to be sent to the p2p network.